libsolidity/lsp/LanguageServer.cpp

```diff
@@ -181,6 +206,18 @@ void LanguageServer::compile()
 	FileRepository oldRepository(m_fileRepository.basePath(), m_fileRepository.includePaths());
 	swap(oldRepository, m_fileRepository);
 
+	// Load all solidity files from project.
+	if (m_analyzeAllFilesFromProject)
+		for (auto const& projectFile: allSolidityFilesFromProject())
```

Comment:
@cameel It's not a full compilation, just parse & analyze, however, I am aware of the fact that we could certainly improve on caching some bits here and there. This is well known and last time we talked about and agreed that we're fine in doing optimizations on that level at a later stage.

IMHO we could re-evaluate that on Q4 (Because until Q3 i have other more user-facing things in the the backlog)

Reply:
Call Summary: we leave analyze-all-files-in-process enabled by default. with respect to perf, I think iff that becomes a problem these users can still disable the functionality, and we eventually have perf opts built in (PR) anyways. But dealing with keeping watch dirs up to date goes way out of the scope of this PR. But a future PR makes sense.